src/transformation-components/Compare.tsx

```diff
@@ -97,112 +85,90 @@ export function Compare({ setErrMsg }: CompareProps): ReactElement {
-      addContextUpdateListener(inputDataContext1, () => {
-        transform(true);
-      });
-      return () => removeContextUpdateListener(inputDataContext1);
-    }
-  }, [transform, inputDataContext1]);
+  useContextUpdateListenerWithFlowEffect(
+    inputDataContext1,
+    lastContextName,
+    () => {
+      transform(true);
+    },
+    [transform]
+  );
 
-  // Listen for updates to second data context
-  useEffect(() => {
-    if (inputDataContext2 !== null) {
-      addContextUpdateListener(inputDataContext2, () => {
-        transform(true);
-      });
-      return () => removeContextUpdateListener(inputDataContext2);
-    }
-  }, [transform, inputDataContext2]);
+  useContextUpdateListenerWithFlowEffect(
+    inputDataContext2,
+    lastContextName,
+    () => {
+      transform(true);
+    },
+    [transform]
+  );
 
   return (
     <>
       <p>Table to Compare 1</p>
-      <select
-        id="inputDataContext1"
+      <CodapFlowSelect
         onChange={inputDataContext1OnChange}
-        defaultValue="default"
-      >
-        <option disabled value="default">
-          Select a Data Context
-        </option>
-        {dataContexts.map((dataContext) => (
-          <option key={dataContext.name} value={dataContext.name}>
-            {dataContext.title} ({dataContext.name})
-          </option>
-        ))}
-      </select>
+        options={dataContexts.map((dataContext) => ({
+          value: dataContext.name,
+          title: dataContext.title,
+        }))}
+        value={inputDataContext1}
+        defaultValue="Select a Data Context"
+      />
       <p>Table to Compare 2</p>
-      <select
-        id="inputDataContext2"
+      <CodapFlowSelect
         onChange={inputDataContext2OnChange}
-        defaultValue="default"
-      >
-        <option disabled value="default">
-          Select a Data Context
-        </option>
-        {dataContexts.map((dataContext) => (
-          <option key={dataContext.name} value={dataContext.name}>
-            {dataContext.title} ({dataContext.name})
-          </option>
-        ))}
-      </select>
+        options={dataContexts.map((dataContext) => ({
+          value: dataContext.name,
+          title: dataContext.title,
+        }))}
+        value={inputDataContext2}
+        defaultValue="Select a Data Context"
+      />
 
       <p>First attribute to Compare</p>
-      <select
-        id="inputAttribute1"
+      <CodapFlowSelect
         onChange={inputAttribute1OnChange}
-        defaultValue="default"
-      >
-        <option disabled value="default">
-          Select a attribute
-        </option>
-        {attributes1.map((attribute) => (
-          <option key={attribute.name} value={attribute.name}>
-            {attribute.title} ({attribute.name})
-          </option>
-        ))}
-      </select>
+        options={attributes1.map((attribute) => ({
+          value: attribute.name,
+          title: `${attribute.title} (${attribute.name})`,
+        }))}
```

Comment:
This title gives an odd doubling of the attribute name because the "title (name)" formatting is also happening in the CodapFlowSelect component. For example when I open this dropdown I see "State (State) (State)". We can probably just omit the formatting here and let CodapFlowSelect be the one to handle it.

See also lines 143-146 below this for the second attribute dropdown.

Reply:
I fixed this in #36 if you want to look at that